operators/advanced_network/python/check_system.py

```diff
@@ -0,0 +1,653 @@
+        return False
+
+
+def check_nvidia_gpu_clocks():
+    """
+    Checks all NVIDIA GPUs to ensure that the SM clock and memory clock are set to their maximum values.
+    If not, prints the current and maximum values for each GPU.
+    """
+    try:
+        # Run nvidia-smi to get clock information
+        result = subprocess.run(
+            ["nvidia-smi", "-q", "-d", "CLOCK"], capture_output=True, text=True, check=True
+        )
+
+        # Parse the output of nvidia-smi
+        output = result.stdout.splitlines()
+
+        current_gpu = None
+        sm_current = None
+        sm_max = None
+        mem_current = None
+        mem_max = None
+
+        for line in output:
+            line = line.strip()
+
+            # Detect GPU identifier
+            if line.startswith("GPU"):
+                current_gpu = line.split()[1].strip(":")
+                sm_current = None
+                sm_max = None
+                mem_current = None
+                mem_max = None
+
+            # Parse current clocks under "Clocks" section
+            elif sm_current is None and line.startswith("SM") and "MHz" in line:
+                sm_current = int(line.split(":")[1].strip().split()[0])
+            elif mem_current is None and line.startswith("Memory") and "MHz" in line:
+                mem_current = int(line.split(":")[1].strip().split()[0])
+
+            # Parse maximum clocks under "Max Clocks" section
+            elif line.startswith("Max Clocks"):
+                continue  # Skip the header for Max Clocks section
+            elif line.startswith("SM") and "MHz" in line and sm_max is None:
+                sm_max = int(line.split(":")[1].strip().split()[0])
+            elif line.startswith("Memory") and "MHz" in line and mem_max is None:
+                mem_max = int(line.split(":")[1].strip().split()[0])
+
+            # Once all data for a GPU is collected, compare clocks
+            if (
+                current_gpu is not None
+                and sm_current is not None
+                and sm_max is not None
+                and mem_current is not None
+                and mem_max is not None
+            ):
+                logging.info(f"GPU {current_gpu}: Checking clocks...")
+
+                if sm_current != sm_max:
+                    logging.warning(
+                        f"GPU {current_gpu}: SM Clock is set to {sm_current} MHz, but should be {sm_max} MHz."
+                    )
+                else:
+                    logging.info(f"GPU {current_gpu}: SM Clock is correctly set to {sm_max} MHz.")
+
+                if mem_current != mem_max:
+                    logging.warning(
+                        f"GPU {current_gpu}: Memory Clock is set to {mem_current} MHz, but should be {mem_max} MHz."
+                    )
+                else:
+                    logging.info(
+                        f"GPU {current_gpu}: Memory Clock is correctly set to {mem_max} MHz."
+                    )
```

Comment:
Would like to see GPU ID and not just address in logging. would be addressed by doing https://github.com/nvidia-holoscan/holohub/pull/608/files#r1876256759